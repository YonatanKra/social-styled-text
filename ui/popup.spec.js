import chrome from 'sinon-chrome';
global.chrome = chrome;

function setButtonInPage() {
    document.body.innerHTML = '<button class="bolderize">Bolderize</button>';
}

function spyOnClickCallback(bolderizeButton) {
    const originalAddEventListener = bolderizeButton.addEventListener;
    const spy = jest.fn();
    jest.spyOn(bolderizeButton, 'addEventListener').mockImplementation((eventName, cb) => {
        spy.mockImplementation(cb);
        originalAddEventListener(eventName, spy);
    });
    return spy;
}

describe(`popup`, function () {
    let bolderizeButton;

    beforeEach(function () {
        setButtonInPage();
        bolderizeButton = document.querySelector('.bolderize');
    });

    afterEach(function () {
        jest.restoreAllMocks();
    });

    it(`should set a listener on the bolderize button`, async function () {
        const spy = spyOnClickCallback(bolderizeButton);
        await import('./popup.js');

        bolderizeButton.click();

        expect(spy).toHaveBeenCalled();
    });

    it(`should work on current tab`, async function () {
        jest.spyOn(chrome.tabs, 'query');

        await import('./popup.js');

        bolderizeButton.click();

        expect(chrome.tabs.query).toHaveBeenCalledWith({active: true, currentWindow: true}, expect.any(Function));
    });

    it('should replace the text of the button with bold text', async function () {
        const textElement = document.createElement('div');
        textElement.innerHTML = '<span>Bolderize</span>';
        document.body.appendChild(textElement);

        await import('./popup.js');
        window.getSelection().selectAllChildren(textElement);
        bolderizeButton.click();

        expect(textElement.textContent).toEqual('𝐁𝐨𝐥𝐝𝐞𝐫𝐢𝐳𝐞');
    });
});
