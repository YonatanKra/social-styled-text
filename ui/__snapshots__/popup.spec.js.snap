// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup should replace the text of the button with bold text 1`] = `"𝗕𝗼𝗹𝗱𝗲𝗿𝗶𝘇𝗲"`;
